src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -438,22 +438,38 @@ std::pair<std::string, mediapipe_packet_type_enum> MediapipeGraphDefinition::get
     return {"", mediapipe_packet_type_enum::UNKNOWN};
 }
 
+struct PythonResourceInitializationCleaningGuard {
+    bool shouldCleanup{true};
+    std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>>& resource;
+    PythonResourceInitializationCleaningGuard(std::unordered_map<std::string, std::shared_ptr<PythonNodeResource>>& resource) :
+        resource(resource) {}
+    ~PythonResourceInitializationCleaningGuard() {
+        if (shouldCleanup) {
+            resource.clear();
+        }
+    }
+    void markSuccess() {
```

Comment:
What about naming this method to something like: `doNotClean`, `cleanupNotNeeded` or `abandonCleanup`, so it more directly tells about the consequences.

Reply:
changed